### PR TITLE
TUP-231: Part 1 of N to Core-UI (Fix Core-CMS / Core-Styles Mismatch)

### PR DESCRIPTION
**Pull request by wesleyboar**
## To Do

- [ ] _After_ approval of this PR, _after_ merge of Core-Styles PR, but _before_ merge of this PR, revert [addition of Core-Styles branch to `devDependencies` in `package.json`](https://github.com/TACC/Core-CMS/pull/482/files#r861999011).

## Overview / Changes

Move back CMS-specific styles to [Core-CMS] repo from [Core-Styles] repo.

<details><summary>Background</summary>

I moved _all_ styles from Core-CMS to Core-Styles. That included some CMS-specific styles. Now I bring them back to CMS.

</details>

## Related

- [TUP-231](https://jira.tacc.utexas.edu/browse/TUP-231)
- required by https://github.com/TACC/Core-Styles/pull/17

## Screenshots

N/A

### Remote

__Confirm server builds and deploys succeed.__

<details>

1. Does website load?
2. Do any stylesheets fail to load?
3. Is UI unchanged? (Compare to prod.)

| Site | Build | Deploy | Server | Note |
| - | - | - | - | - |
| Core | [#618](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/618) |  [#1395](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1395) | [Dev](http://dev.cep.tacc.utexas.edu/) | |
| A2CPS | [#619](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/619) | [#1396](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1396) | [Pre-Prod](https://pprd.a2cps.tacc.utexas.edu/) | |
| Brainmap | [#620](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/620) | [#1397](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1397) | [Pre-Prod](https://pprd.brainmap.tacc.utexas.edu/about) | |
| ECEP | [#621](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/621) | [#1398](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1398) | [Pre-Prod](https://pprd.ecep.tacc.utexas.edu/) | |
| Frontera | [#622](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/622) | ~~[#1399](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1399)~~ ~~[#1402](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1402)~~ | ~~[Pre-Prod](https://pprd.frontera-portal.tacc.utexas.edu/)~~ | deploy fail, time out |
| Neuronex | [#623](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/623) | [#1400](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1400) | [Pre-Prod](https://3dem-staging.tacc.utexas.edu/) | |
| ProTX | [#624](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/624) | — | — | waiting for unfork |
| SciVisColor | [#625](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/625) | — | — | no Camino deploy |
| UTRC | [#626](https://jenkins01.tacc.utexas.edu/job/Core_CMS_Build/626) | [#1401](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1401) | [Pre-Prod](https://pprd.utrc.tacc.utexas.edu/) | |

</details>

### Local

__1. Confirm _near_-identical build output for project(s) on _`main`_  branch versus _this_ branch for different projects.__

<details>

Follow [How to Test Major Changes to CSS](https://github.com/TACC/Core-CMS/wiki/How-to-Test-Major-Changes-to-CSS). The expected differences in build output are:

| File | `;` → `}:root{` | `font-family` _before_ `font-size` → `font-family` _after_ `font-weight` |
| - | - | - |
| `site.css` | ✅  | ✅  |
| `site.header.css` | ❌  | ✅  |
| `site.header.docs.css` | ✅  | ❌  |

Those differences occur because:

| Difference | Reason |
| - | - |
| `;` → `}:root{` | Properties that were in one `:root` ruleset in one file have been moved into a `:root` ruleset for each of two files. |
| `font-family` _before_ `font-size` → `font-family` _after_ `font-weight` | The `font-` properties for Core-CMS (`font-family`) now load _after_ the Core-Styles properties (`font-size` and `font-weight`). |

_You don't need to test build _output_ of each project, because zero project stylesheets load the files **changed** in this branch on Core-CMS and Core-Styles._

</details>

__2. Confirm all projects[^1] builds _with no errors_.__

<details>

Run `npm run build --project=____` for each project, where `____` is the project directory in `Core-CMS-Resources`.

[Core-CMS]: https://github.com/TACC/Core-CMS
[Core-Styles]: https://github.com/TACC/Core-Styles

_A build failure would be caused by a bad CSS import path._

</details>

[^1]: Except LCCF, cuz it has an existing bug; it needs a placeholder stylesheet; ultra low priority.

### CODE REVIEW

File: taccsite_cms/static/site_cms/css/src/_imports/generics/fonts.css [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/generics/fonts.css:1]: (copied _entire file_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/generics/fonts.css))

File: taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.post.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.post.css:1]: (copied _entire file_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/components/django.cms.post.css))

File: taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/*
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.css:1]: (copied _entire file_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/components/django.cms.css))

File: taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.blog.css [hunk 4]
```diff
@@ -0,0 +1,11 @@
+/*
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/components/django.cms.blog.css:1]: (copied _entire file_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/components/django.cms.blog.css))

File: taccsite_cms/static/site_cms/css/src/_imports/settings/color.css [hunk 5]
```diff
@@ -0,0 +1,10 @@
+:root {
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/settings/color.css:1]: (copied _CMS-only properties_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/settings/color.css))

File: taccsite_cms/static/site_cms/css/src/_imports/settings/font.css [hunk 6]
```diff
@@ -0,0 +1,12 @@
+:root {
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/settings/font.css:1]: (copied _CMS-only properties_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/settings/font.css))

File: taccsite_cms/static/site_cms/css/src/_imports/settings/space.css [hunk 7]
```diff
@@ -0,0 +1,3 @@
+:root {
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/settings/space.css:1]: (copied _CMS-only properties_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/settings/props.space.css))

File: taccsite_cms/static/site_cms/css/src/site.css [hunk 8]
```diff
@@ -4,16 +4,19 @@
...
 @import url("_imports/settings/color.css");
-@import url("_imports/settings/props.max-width.css");
-@import url("_imports/settings/props.space.css");
 @import url("_imports/settings/font.css");
+@import url("_imports/settings/space.css");
+@import url("_imports/settings/max-width.css");
+/* Unique to CMS */
+@import url("./_imports/settings/color.css");
+@import url("./_imports/settings/font.css");
+@import url("./_imports/settings/space.css");
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/site.css:15]: (removed `props` filename prefix in TACC/Core-Styles#17 [`…max-width.css`](https://github.com/TACC/Core-Styles/pull/17/files#diff-0d662592406f132365a8223f05a1af57f24674325a1aed265e53136ed8819f70) and [`…space.css`](https://github.com/TACC/Core-Styles/pull/17/files#diff-05ef9222be1c3e7e67c68e1fce3c5dae9b3122e93bb8cba09bee8f7daadcb48a))

File: taccsite_cms/static/site_cms/css/src/site.css [hunk 9]
```diff
@@ -27,19 +30,18 @@
 @import url("_imports/objects/o-site.css");
 
 /* COMPONENTS */
-/* GH-302: HELP: How should all of these become individually built files? */
-/* GH-302: FAQ: Individually built stylesheets could be loaded explicitely. */
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/site.css:31]: This is a possible task for Core-Styles now, not Core-CMS.

File: taccsite_cms/templates/nav_search.raw.html [hunk 10]
```diff
@@ -3,9 +3,7 @@
           so Portal and User Guide can render this markup into their markup. -->
 
 <style type="text/css">
-  /* Hide search bar before all relevant styles have loaded */
-  /* FAQ: Manage visibility of search bar to avoid FOUC */
-  /* HACK: GH-101: Resolve via dynamically-added `[data-status="loaded"]` */
+  /* SEE: https://github.com/TACC/Core-Styles/blob/main/source/_imports/elements/tacc-search-bar.md */
```
Comment (wesleyboar) [inline taccsite_cms/templates/nav_search.raw.html:6]: That _will_ be the URL. Until https://github.com/TACC/Core-Styles/pull/17, the actual URL is https://github.com/TACC/Core-Styles/blob/task/TUP-231-cms-css-from-styles-repo-to-cms-repo/source/_imports/elements/tacc-search-bar.md.

**Review by wesleyboar — COMMENTED** [review]
notes for reviewers

File: package.json [hunk 11]
```diff
@@ -9,7 +9,7 @@
     "npm": "8.5.5"
   },
   "devDependencies": {
-    "@tacc/core-styles": "git+https://git@github.com/TACC/Core-Styles.git",
+    "@tacc/core-styles": "git+https://git@github.com/TACC/Core-Styles.git#task/TUP-231-cms-css-from-styles-repo-to-cms-repo",
```
Comment (wesleyboar) [inline package.json:12]: **Warning!** This change should **not** be committed. It is required only for testing (remote and local).

File: taccsite_cms/static/site_cms/css/src/_imports/settings/font.css [hunk 12]
```diff
@@ -0,0 +1,14 @@
+@import url("@tacc/core-styles/source/_imports/settings/font.css");
+
+:root {
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/_imports/settings/font.css:3]: (copied _CMS-only properties_ from [TACC/Core-Styles:/source/...](https://github.com/TACC/Core-Styles/blob/b70417e/source/_imports/settings/fonts.css))

File: taccsite_cms/static/site_cms/css/src/site.css [hunk 13]
```diff
@@ -4,16 +4,15 @@
...
 @import url("_imports/settings/border.css");
-/* Unique to CMS */
-@import url("_imports/settings/color.css");
-@import url("_imports/settings/props.max-width.css");
-@import url("_imports/settings/props.space.css");
-@import url("_imports/settings/font.css");
+/* Unique to CMS */    @import url("./_imports/settings/color.css");
+@import url("_imports/settings/max-width.css");
+/* Unique to CMS */    @import url("./_imports/settings/space.css");
+/* Unique to CMS */    @import url("./_imports/settings/font.css");
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/css/src/site.css:11]: **Note.** I did not re-order these, _only_ so that local testing—comparing build output—is easier. I would like to, because the order is actually irrelevant to the build, but important for legibility.

Comment (taoteg) [reply taccsite_cms/static/site_cms/css/src/site.css:11]: I can see that. You want to make that a post-merge task? Or do it now?

**Review by taoteg — APPROVED** [review]
LGTM!

File: taccsite_cms/static/site_cms/css/src/site.css (hunk 13 above)
Comment (wesleyboar) [reply taccsite_cms/static/site_cms/css/src/site.css:11]: Ugh. I want to do it now, but that'll make my test notes inaccurate, so I'll do it in a tiny post-merge PR.

**Comment by wesleyboar** [comment]
I merged in `main` to this branch, since the merge of #466. Then I followed local test instructions this branch using `texascale-org` and `ecep-cms` (the projects relevant to #466). Alls well. Merging, now.

File: taccsite_cms/static/site_cms/css/src/site.css (hunk 13 above)
Comment (wesleyboar) [reply taccsite_cms/static/site_cms/css/src/site.css:11]: K, legibility improved in follow-up PR merged) [#484](https://github.com/TACC/Core-CMS/pull/484/files).
